### PR TITLE
owner: fix a bug which lead to replication stopped and no error report

### PR DESCRIPTION
**Pull request by leoppro**
### Bug phenomenon

The resolved TS and checkpoint TS of a change feed is stopped, but no error reported

### How to confirm the bug
We can check the owner log, and find logs like this:
```

[2021/04/25 21:39:48.899 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":694,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:48.899 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":1103,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:48.989 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [capture-id=1e8e5271-9340-4b08-b0b4-a00dd456eda6] [status="{\"tables\":{\"1103\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"312\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"452\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"694\":{\"start-ts\":424504976445538305,\"mark-table-id\":0}},\"operation\":null,\"admin-job-type\":0}"]
[2021/04/25 21:39:48.989 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":694,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:48.989 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":1103,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:49.049 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [capture-id=1e8e5271-9340-4b08-b0b4-a00dd456eda6] [status="{\"tables\":{\"1103\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"312\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"452\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"694\":{\"start-ts\":424504976445538305,\"mark-table-id\":0}},\"operation\":null,\"admin-job-type\":0}"]
[2021/04/25 21:39:49.049 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":1103,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:49.049 +08:00] [WARN] [changefeed.go:561] ["the target capture is not exist, sent the table to orphanTables"] [job="{\"From\":\"b51b0e86-373c-49a2-a7c4-42c98fde4e5e\",\"To\":\"5ec78795-88ea-43b3-803b-229a3ededc3b\",\"TableID\":694,\"TableReplicaInfo\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"Status\":1}"]
[2021/04/25 21:39:49.111 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [capture-id=1e8e5271-9340-4b08-b0b4-a00dd456eda6] [status="{\"tables\":{\"1103\":{\"start-ts\":424504976445538305,\"mark-table-id\":0},\"312\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"452\":{\"start-ts\":424504976340680705,\"mark-table-id\":0},\"694\":{\"start-ts\":424504976445538305,\"mark-table-id\":0}},\"operation\":null,\"admin-job-type\":0}"]
```
### Trigger conditions

A capture is offline when some table is moving by the owner.

### Versions have this bug
[4.0.0, 4.0.13], 5.0.0-rc, [5.0.0, 5.0.1]

### Bug mechanism

when the owner moves a table, the owner tries to remove the table from source capture and add the table to target capture.
if the target capture is offline at the same time, the owner should add this table to orphan tables and dispatch this table in the next tick.

but the owner forget to remove the invalid move table job, so the owner will add the table to orphan table every tick, leads to the other logic does not work properly

### Check List 

Tests 

 - Unit test

### Release note

- fix a bug about resolved ts stopped when move a table.

### CODE REVIEW

File: cdc/changefeed_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	s.cancel()
+	err := s.errg.Wait()
+	if err != nil {
+		c.Errorf("Error group error: %s", err)
+	}
+	s.client.Close() //nolint:errcheck
+}
+
+func (s *changefeedSuite) TestHandleMoveTableJobs(c *check.C) {
+	defer testleak.AfterTest(c)()
```
Comment (amyangfei) [inline cdc/changefeed_test.go:74]: we should call `s. TearDownTest(c)` to release resources before leak test

Comment (leoppro) [reply cdc/changefeed_test.go:74]: fixed

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Review by amyangfei — APPROVED** [review]

**Pull request merged**